recipes/cpprestsdk/all/conanfile.py

```diff
@@ -62,57 +51,61 @@ def configure(self):
             del self.options.fPIC
 
     def requirements(self):
-        self.requires("boost/1.79.0")
-        self.requires("openssl/1.1.1o")
+        self.requires("boost/1.80.0")
+        self.requires("openssl/1.1.1s")
         if self.options.with_compression:
-            self.requires("zlib/1.2.12")
+            self.requires("zlib/1.2.13")
         if self.options.with_websockets:
             self.requires("websocketpp/0.8.2")
 
     def package_id(self):
         self.info.requires["boost"].minor_mode()
 
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
-
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
-
-        self._cmake = CMake(self, set_cmake_flags=True)
-        self._cmake.definitions["BUILD_TESTS"] = False
-        self._cmake.definitions["BUILD_SAMPLES"] = False
-        self._cmake.definitions["WERROR"] = False
-        self._cmake.definitions["CPPREST_EXCLUDE_WEBSOCKETS"] = not self.options.with_websockets
-        self._cmake.definitions["CPPREST_EXCLUDE_COMPRESSION"] = not self.options.with_compression
+        files.get(self, **self.conan_data["sources"][self.version],
+                  destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        # upstream CMakeLists.txt sets BUILD_SHARED_LIBS as a CACHE variable
+        # remove for conan >=1.54.0
+        tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0126"] = "NEW"
```

Comment:
CMP0126 NEW can't be guaranteed since it requires CMake >= 3.21, and conan-center supports CMake >= 3.15

Reply:
It's also fixed by Conan 1.54.0

Reply:
yes, meanwhile here is a fix: https://github.com/conan-io/conan-center-index/pull/14227

Reply:
I hope you're right! I thought I'd tried injecting the non-cache variable... and per the CMake documentation https://cmake.org/cmake/help/v3.20/command/set.html "any normal variable binding in the current scope will be removed to expose the newly cached value".

Reply:
Actually tc.variables name is misleading, it injects a CACHE variable in CMakeToolchain, which tc.cache_variables injects a cache_variable in CMakePresets & in cmake configuration command line when called afterwards by CMake build helper configure() method.